backend/src/auth/controller/auth.controller.ts

```diff
@@ -105,14 +107,21 @@ export class AuthController {
     return await this.authService.handleAppleOAuthLogin(req);
   }
 
-  @Public()
-  @Get('search')
-  async searchUsers(@Query('searchTerm') searchTerm: string) {
-    if (!searchTerm) {
-      throw new HttpException('Search term is required.', HttpStatus.BAD_REQUEST);
+  // New API endpoint for social account linking
+  @Post('/link-social')
```

Comment:
I feel the best place for this would be under `user.controller.ts` since the endpoint is more user specific!